plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -409,7 +409,16 @@ def process_main(self, item=None):
             if not checksum_type:
                 # Never assume. But oh well
                 checksum_type = "sha256"
-            blob_src_path = os.path.join(self.get_working_dir(), checksum + '.tar')
+
+            blob_src_path = os.path.join(self.get_working_dir(), checksum)
+            version_file_path = os.path.join(self.get_working_dir(), 'version')
```

Comment:
i was wondering if it is a good idea to rely on the version file because of the future rfe https://pulp.plan.io/issues/3497  where we'd allow to upload to pulp partial tar that would not contain whole content.

what do you think about using glob? we can provide a pathname pattern
```
In [11]: glob.glob(dir + checksum + '.tar')
Out[11]: ['/home/ipanova/pulp_development/pulp_docker/existingemptydirectory/8c811b4aec35f259572d0f79207bc0678df4c736eeec50bc9fec37ed936a472a.tar']
```
in case it'd find a blob with tar extension it would return a list otherwise it would be empty

Reply:
IMO, I would still use the version file. Remember that the user is not uploading skopeo output directly but is instead being forced to tar it up. If I were a smart user, I'd write a script that made the tar files optimal to not duplicate layers. Also consider that a future version of skopeo may change the format of the output. My vote is to use the version file.

Reply:
I was assuming that for 3497 the tar would still contain a version file. Referenced blobs don't exist, they would be queried in pulp, and I assume the file extension wouldn't matter for that case.

If we are going to support multiple versions of the transport, I think its probably better for the code to be aware of the transport version rather than go with whatever is there.